### ISSUE TITLE
react/jsx-no-undef consistency

### ISSUE THREAD

**Issue by passariello** [oxc-project/oxc#3319]
Hello, recently I move from ESlint to OXC. I figured out an issue about "react/jsx-no-undef" that not record the "0" or "off" or false
I need to stop the warning.

I have a global function (please, don't ask...) and previously works fine.... also "elem"
In substance any globals created into the project going to give me a warning is used as components. 

Error:
"eslint-plugin-react(jsx-no-undef): Disallow undeclared variables in JSX
help: 'dispatch' is not defined. oxc"

I try to fix via the .d.ts but not lucky here! 

Many thanks.
Dario Passariello

![image](https://github.com/oxc-project/oxc/assets/17809213/2a865db0-c51c-43d7-bc22-a47d19f15f44)

**Comment by passariello** [comment]
I try also to use https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-undef.md to define the eslintrc.json but nothing.

**Comment by Boshen** [comment]
Thank you for the feedback, I'll update the rule with these options. https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-undef.md#rule-options

**Comment by passariello** [comment]
Many thanks, I really like the project and simplify alot of things! Thanks.

**Comment by Boshen** [comment]
In the next version: 

In your `eslintrc.json`:

```javascript
{
  "globals": {
    "dispatcher": "readonly"
  }
}
```
